### PR TITLE
Fixes lr-policy for egress gw

### PR DESCRIPTION
**Pull request by trozet**
Pods configured with egress gws were unable to reach other pods in other
subnets, via service or otherwise. This is because the lr-policy for
routing traffic to GR was not specific enough. This patch adds criteria
for matching not cluster subnet, so only external traffic is forwarded
to GR.

Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Comment by trozet** [comment]
@dcbw @rcarrillocruz PTAL

File: go-controller/pkg/ovn/egressgw.go [hunk 1]
```diff
@@ -349,7 +349,13 @@ func (oc *Controller) addHybridRoutePolicyForPod(podIP net.IP, node string) erro
 		if err != nil {
 			return fmt.Errorf("failed to parse gateway router join interface IP: %s, err: %v", grJoinIP, err)
 		}
+		var matchDst string
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
+			matchDst += fmt.Sprintf(" && %s.dst != %s", l3Prefix, clusterSubnet.CIDR)
```
Comment (aojea) [inline go-controller/pkg/ovn/egressgw.go:354]: we don' t have any test that checks these expressions?

Comment (trozet) [reply go-controller/pkg/ovn/egressgw.go:354]: We have e2e control plane tests that verify egress gw works. However those test do not validate pod to service (regular k8s) traffic on egress gw pods. We should add a test for this @nerdalert

Comment (danwinship) [inline go-controller/pkg/ovn/egressgw.go:354]: as with the other PR this needs to check `utilnet.isIPv6CIDR(clusterSubnet.CIDR)` to handle dual-stack (and again below when deleting)

**Review by danwinship — COMMENTED** [review]
I don't know this code but from your description the patch sounds right

File: go-controller/pkg/ovn/egressgw.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/egressgw.go:354]: will fix

Comment (rcarrillocruz) [reply go-controller/pkg/ovn/egressgw.go:354]: Actually, I should take that.
I pulled that work from @nerdalert .

**Comment by danwinship** [comment]
lgtm

**Pull request merged**